File: device_simulator/message_request.py
class MessageRequest:
    def __init__(self, text: str, date_time: str, dev_id: int, dev_type: str):
        self.message_text: str = text
        self.date_time: str = date_time
        self.dev_id: int = dev_id
        self.dev_type: str = dev_type

    def get_json(self):
        return {
            "messageText": self.message_text,
            "dateTime": self.date_time,
            "deviceId": self.dev_id,
            "deviceType": self.dev_type
        }

File: smart-home-backend/src/main/java/com/ftn/security/smarthomebackend/dto/request/MessageRequest.java
package com.ftn.security.smarthomebackend.dto.request;

import com.ftn.security.smarthomebackend.enums.DeviceType;
import jakarta.persistence.Column;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
public class MessageRequest {

    @NotBlank(message="Message can not be empty.")
    private String messageText;

    @Future(message="Date time can not be in future.")
    private LocalDateTime dateTime;

    @NotNull(message="Message have to have device id.")
    private Long deviceId;

    @NotNull(message="Message have to have device type.")
    private DeviceType deviceType;

    public MessageRequest(String messageText, LocalDateTime dateTime, Long deviceId, DeviceType deviceType) {
        this.messageText = messageText;
        this.dateTime = dateTime;
        this.deviceId = deviceId;
        this.deviceType = deviceType;
    }
}
